cmd/pmemctl/pmemctl.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+FILE=""
+FS=""
+MNT_DIR=""
+SIZE=""
+BLOCK_SIZE=4096
+
+die() {
+	echo "ERROR: $*" >&2
+	exit 1
+}
+
+warn() {
+	echo -e "WARNING: $*" >&2
+}
+
+info() {
+	echo -e "INFO: $*"
+}
+
+help() {
+cat << EOF
+$0 is a command line tool to create raw files that can be used as pmem
+volumes that support DAX[1] in Kata Containers
+
+USAGE:
+	$0 [OPTIONS] FILE
+
+EXAMPLE:
+	$0 -s 10G -f xfs -m ~/dax_volume ~/file.dax
+
+OPTIONS:
+	-f FS   filesystem, only xfs(5) and ext4(5) support DAX[1]
+	-m DIR  mount point in the host. Create a loop device for the file
+	        and mount it in the DIR specified, DIR can be used as a volume
+	        and Kata Containers will use it as a pmem volume that support DAX
+	-s SIZE file size. SIZE must be aligned to 128M, hence the minimum
+	        supported size for a file is 128M
+
+[1]- https://www.kernel.org/doc/Documentation/filesystems/dax.txt
+EOF
+}
+
+create_file() {
+	set -x
+	local reserved_blocks_percentage=3
+	local block_size=4096
+	local data_offset=$((1024*1024*2))
+	local dax_alignment=$((1024*1024*2))
+	local nsdax_src_url="https://raw.githubusercontent.com/kata-containers/osbuilder/master/image-builder/nsdax.gpl.c"
+
+	truncate -s "${SIZE}" "${FILE}"
+	if [ ! -x nsdax ]; then
+		curl -s "${nsdax_src_url}" | gcc -x c -o nsdax -
```

Comment:
- `curl -Ls` might be safer.
- That is one crazy cool command line! :smile:
- Since we don't want users running network operations like this as `root`, it might be worth checking to ensure the user runs this script as a non-priv user. You are using `sudo` consistently so that should be fine.